inlong-sort/sort-connectors/iceberg/src/main/java/org/apache/inlong/sort/iceberg/sink/multiple/IcebergSingleStreamWriter.java

```diff
@@ -125,37 +114,7 @@ public void prepareSnapshotPreBarrier(long checkpointId) throws Exception {
 
     @Override
     public void processElement(T value) throws Exception {
-        try {
-            writer.write(value);
-        } catch (Exception e) {
-            LOGGER.error(String.format("write error, raw data: %s", value), e);
-            if (!dirtyOptions.ignoreDirty()) {
-                throw e;
-            }
-            if (dirtySink != null) {
-                DirtyData.Builder<Object> builder = DirtyData.builder();
-                try {
-                    builder.setData(value)
-                            .setLabels(dirtyOptions.getLabels())
-                            .setLogTag(dirtyOptions.getLogTag())
-                            .setIdentifier(dirtyOptions.getIdentifier())
-                            .setRowType(flinkRowType)
-                            .setDirtyMessage(e.getMessage());
-                    dirtySink.invoke(builder.build());
-                    if (metricData != null) {
-                        metricData.invokeDirtyWithEstimate(value);
-                    }
-                } catch (Exception ex) {
-                    if (!dirtyOptions.ignoreSideOutputErrors()) {
-                        throw new RuntimeException(ex);
-                    }
-                    LOGGER.warn("Dirty sink failed", ex);
```

Comment:
Maybe this will cause single table sync lost dirty data process when remove it

Reply:
The operations on metric and dirtySink are moved to the following methods:
IcebergMultipleStreamWriter#processElement

Reply:
Compared with SinkMetricData, SinkTableMetricData not only collects the metric of each table, but also summarizes the metric of a node;
So the SinkTableMetricData object is moved to the IcebergMultipleStreamWriter class, and the IcebergSingleStreamWriter only corresponds to a single table;

Reply:
The operations on metric and dirtySink have been moved to the following methods:
IcebergMultipleStreamWriter#processElement

Reply:
> Compared with SinkMetricData, SinkTableMetricData not only collects the metric of each table, but also summarizes the metric of a node; So the SinkTableMetricData object is moved to the IcebergMultipleStreamWriter class, and the IcebergSingleStreamWriter only corresponds to a single table;

@lordcheng10 single table sync will exec FlinkSink#chainIcebergSingleSinkOperators#createStreamWriter#IcebergSingleStreamWriter. So metric will and dirty data process logic will lost